packages/utils/parcelforvscode/vscode-extension-TODO.md

```diff
@@ -0,0 +1,32 @@
+Packages:
+
+- [@parcel/reporter-lsp](./packages/reporters/lsp-reporter/)
+- [parcel-for-vscode](./packages/utils/parcelforvscode/)
+- [@parcel/lsp](./packages/utils/parcel-lsp/)
+- [@parcel/lsp-protocol](./packages/utils/parcel-lsp-protocol)
+
+TODO:
+
+- [x] need to not wait for connections
+- [x] language server shuts down and kills our process when the extension is closed
+- [x] handle the case where parcel is started after the extension is running
+- [x] handle the case where extension is started while parcel is running
+- [x] support multiple parcels
+- [x] show prior diagnostics on connection
+- [x] only connect to parcels that match the workspace
+- [ ] show parcel diagnostic hints
+- [ ] implement quick fixes (requires Parcel changes?)
+- [x] cleanup LSP server sentinel when server shuts down
```

Comment:
Is this done? How does the sentinel file get cleaned up?

Reply:
It's cleaned up by `client.onClose` in `LspServer.js`

Reply:
🤔 Maybe i'm missing something, but I don't see anywhere where the sentinel file is unlinked. AFAICT, the sentinel file will just exist in the tmp dir forever once the lsp server has been started once.

If that's the case, i think it's fine for now. We can just uncheck this task to remind ourselves we still need to do it.

We might want to enforce a singleton condition on the lsp server, as well, so the sentinel might have change a bit, anyway.

Reply:
Added [this](https://github.com/parcel-bundler/parcel/pull/9278/files#diff-f3c6654ce2aa9ac5aa019e5af5506352169f4453e99e57ef2dba06aeab2f81ccR55) to the `deactivate` function in `extension.ts`. I think we should only remove the sentinel file when the extension is deactivated.